### DISCUSSION TITLE
syncing assignments

### DISCUSSION THREAD

**Discussion by rossetti** [community/community#136584] [GitHub Education]
### Hi everyone,

Has anybody experienced this error?  How do you contact support?

Error
Failed to open 60 of 60 pull requests. This may mean that pull requests to sync your changes have already been successfully opened to your students repositories. If this is not the case, please try again or contact support.

**Comment by jijones-clayton-state** [comment]
I've gotten it for a few days now, but the PRs are still created for the students, so I don't know what's going on with it. I thought it was because a few of my students were behind on merging the PR, and that may be the case since today, I got this error:
![image](https://github.com/user-attachments/assets/bcc3bd7f-f9f9-47a7-b7e3-d12fc7664100)

I am 90% sure that those 10 students have not merged the last PR. Could that be the issue? It's expecting that GitHub Classroom: Sync Assignment PR to not be there, but since it is, it throws an error.

I found a way to put in a ticket and have done so just now, so hopefully we hear back from someone soon!

**Comment by nadia-eecs** [comment]
Unable to sync student repositories after updating the template assignment repository with a non-conflicting file add after students accepted assignment. 441 student classroom size with 180 students accepting assignment prior to the addition of the file. Of the 180 repositories, 80 seem to have synced based on the message. Sync failed 10/6/2024, and re-sync was attempted 10/7/2024. Contacting support for assistance didn't help much so I am posting here to see if someone from the community may assist me.

![image](https://github.com/user-attachments/assets/c194ffa4-d5a9-4490-874a-ebc87290d59b)

Any help will be of great value!

**Reply by rossetti** [reply]
Unfortunately, I have continued to experience this issue on any of the homework assignments in my class. I have basically given up on github classroom for this semester and shared my class code via a github organization.  I would love to know how to fix this issue in the future.

**Reply by jijones-clayton-state** [reply]
I'm hoping that the support people sent our tickets over to @RyanHecht and his team.

I know that, for mine, some of the pull requests were created and I think I pinpointed the problem with mine since it was only a partial number this time. I created the assignment weeks ago, then synced an update. I showed them how to get to it in class and expected the majority of them to do it right then and there. I don't think that was the case since I got this message also on my second update. Monday, I did a third one to fix an issue with them not seeing an image, which is when it said 10 of 29. 

Looking at a few of my known stragglers (people who come late to class, etc.), I noticed the old PR from at least 1 week ago is still open and not merged, so I suspect that's what the issue is for my group. I could manually go in and accept it for them, but I'd have to know who was behind first and I don't usually go to the repo itself unless they've closed/merged the Feedback one (which seems to be what a few of them did instead of the Sync one).

A nice thing would be to see (apart from looking at the last update, which, unfortunately, includes the last time the repo was synced) would be who hasn't merged the PR without having to look at each individual one, so then I could do it manually if necessary or at least reach out to them to say that they'll be behind if they don't start working.

**Reply by agutf** [reply]
> Unable to sync student repositories after updating the template assignment repository with a non-conflicting file add after students accepted assignment. 441 student classroom size with 180 students accepting assignment prior to the addition of the file. Of the 180 repositories, 80 seem to have synced based on the message. Sync failed 10/6/2024, and re-sync was attempted 10/7/2024. Contacting support for assistance didn't help much so I am posting here to see if someone from the community may assist me.
> 
> ![image](https://private-user-images.githubusercontent.com/63071784/374392834-c194ffa4-d5a9-4490-874a-ebc87290d59b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkwNjM0ODcsIm5iZiI6MTcyOTA2MzE4NywicGF0aCI6Ii82MzA3MTc4NC8zNzQzOTI4MzQtYzE5NGZmYTQtZDVhOS00NDkwLTg3NGEtZWJjODcyOTBkNTliLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDE2VDA3MTk0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVmNWFlMmRiZmZjMTE1MmI5MDNjZTBmYWVjZTIxN2YwNjI4NTI2MTIxZTMyMWNlYjE2NTA1Zjc3NWM1NzBhZWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.coRUt9lKuvO4IoUFO7wn85R-t0jUPv_iGXJ-_Z6pJKI)
> 
> Any help will be of great value!

Exact same problem here... It would be good if we could be given a way to solve the problem

**Reply by jijones-clayton-state** [reply]
I heard back from support with my issue (where I put out updates before they accepted the previous ones) which they said they'd have to look into it and had no ETA on how long it would take, but told me to have the students rebase their repos manually.

I don't know if that would help you guys or not, but thought I'd share it anyhow.

**Comment by sgoggins** [comment]
I am experiencing this issue. I updated the README.md file in the template repository, but when I go to sync the student repositories either from a student repository or within GitHub classrooms, it tells me in error that the forks are already sync'd and there are no changes. I have quintuple (at least) confirmed that the changes I made in the template repository do in fact exist. 

This is frustrating since the most recent commit hashes are OBVIOUSLY different in the template repository.

**Comment by tturin1017** [comment]
![image](https://github.com/user-attachments/assets/e0d32332-b89a-45a1-bbc5-814f300ccb9d)

Hi , I am having the same issue as everyone. I have two class sections and github behaves differently when attempting to update starter code.. despite starter templates being identical. 
![image](https://github.com/user-attachments/assets/4099ce61-0044-4683-b7bc-d7d3439817cb)
The pull request shows up for one of my sections but not the other. I am new to github classrooms and do not know how to fix. 

Thanks!

**Reply by jijones-clayton-state** [reply]
This is good to know for multiple sections as I have 2 sections this semester, though I think I was going to forego forking this semester because of all the issues.
I wonder what happens if you merge one of the PRs. I wonder if your error message will change to 31 of 32.

**Comment by joel-porquet** [comment]
This issue is a major one. I'm encountering the same problem. I added a file to my starter code and when I try sync'ing with my students' repos, I'm faced with "Unable to sync repositories". Only a few random students get the PR, but most don't.

It's my first quarter trying out GH Classroom, and so far, there are too many issues to be useful in a realistic context.

**Comment by daisla** [comment]
I also regularly get the error "_Unable to sync student repositories. Please try again or contact support._" whenever pushing changes. Generally when I get this, 40-50% of the students end up getting a PR properly generated, the rest ... nothing. I end up going in and generating the PRs by hand, which is super tedious for a class of 80.

Come on, GitHub, this is ridiculous. It's a basic piece of functionality that just doesn't work.

**Comment by Venkat5674** [comment]
### ✅ Explanation & Resolution for “Failed to open pull requests” Error

This error usually occurs when GitHub has **already created the pull requests** (fully or partially) during a previous sync attempt, even if the UI reports a failure.

In most cases, **nothing is actually broken**.

#### Why this happens
- A previous sync may have succeeded, but the page timed out or was refreshed.
- Some or all pull requests already exist in student repositories, and GitHub prevents duplicate PRs.
- Large classrooms (many repos) can hit background job or rate-limit delays.
- Students may have modified or deleted the expected default branch.

#### What to do
1. **Manually check a few student repositories**  
   Open any student repo → go to **Pull Requests**  
   If you see a PR from the assignment or sync branch, the operation was successful.

2. **Wait and retry once**
   GitHub Classroom background jobs can take time.
   Waiting 10–15 minutes before retrying often resolves the issue.

3. **Retry in an incognito/private window**
   This avoids cached session issues that can cause false failures.

4. **Do not force re-sync repeatedly**
   Repeated attempts won’t recreate PRs and may keep showing the same message.

#### When to contact support
If **no pull requests exist in any student repositories**, contact GitHub Support and ask them to **reset the sync state** for the assignment.

Support link:  
https://support.github.com/contact  
(Category: GitHub Classroom / Education)

#### Summary
- This message usually means PRs already exist
- Verify student repos before retrying
- Safe to ignore if PRs are present
- Support can manually fix true sync failures

**Reply by joel-porquet** [reply]
Ok, so that was clearly generated by ChatGPT. And it's quite unhelpful. What's the point of doing that?

> In most cases, nothing is actually broken.

Most people are reporting that some of the PRs go through and some don't. This is clearly broken behavior.

**Reply by samus-aran** [reply]
Hi @Venkat5674 , 

We’ve clarified our stance on using generative AI tools like ChatGPT within our Community via [this announcement](https://github.com/orgs/community/discussions/69006). Please review the guidelines to ensure your post meets them as failure to adhere to those rules can result in action taken by our moderator team.  You can read our updated [Code of Conduct](https://github.com/community/community/blob/baa5cd9020ef3f2bbf86dca1191d9c90f9b0822d/CODE_OF_CONDUCT.md#reasonable-use-of-ai-generated-content) and the [announcement](https://github.com/orgs/community/discussions/69006) for more details. Thank you for helping us maintain an authentic and beneficial space for everyone.
